geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/KryoLazyDensityUtils.scala

```diff
@@ -18,20 +18,22 @@ import org.locationtech.geomesa.utils.geotools.GridSnap
 import org.locationtech.geomesa.utils.interop.SimpleFeatureTypes
 import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 import org.opengis.filter.expression.Expression
+import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
 
 import scala.collection.mutable
 
 trait KryoLazyDensityUtils {
 
   import KryoLazyDensityUtils._
-  var geomIndex: Int = -1
-  // we snap each point into a pixel and aggregate based on that
-  var gridSnap: GridSnap = null
-  var writeGeom: (SimpleFeature, DensityResult) => Unit = null
+
+  var geomIndex: Int = _
+  var gridSnap: GridSnap = _
+  var writeGeom: (SimpleFeature, DensityResult) => Unit = _
 
   def initialize(options: Map[String, String], sft: SimpleFeatureType): DensityResult = {
-    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
-    geomIndex = sft.getGeomIndex
+
+    geomIndex = Option(sft.getGeomIndex).getOrElse(-1)
```

Comment:
think we should revert all the changes to this file too - this isn't going to do quite what you expect, since it's an Int and can't be null, so it will get turned to 0 and thus always be Some. The other changes seem like they're not needed any more